src/symbol/glyph_source.js

```diff
@@ -89,40 +105,39 @@ class GlyphSource extends Evented {
             /* eslint-enable new-cap */
         };
 
-        for (let i = 0; i < glyphIDs.length; i++) {
-            getGlyph(glyphIDs[i]);
+        for (const glyphID of glyphIDs) {
+            getGlyph(glyphID);
         }
 
         if (!remaining) callback(undefined, glyphs, fontstack);
 
-        const onRangeLoaded = (err, range, data) => {
+        const onRangeLoaded = (err: ?Error, range: ?number, data: ?Glyphs) => {
             if (err) {
                 this.fire('error', { error: err });
-                return;
-            }
+            } else if (typeof range === 'number' && data) {
+                const stack = this.stacks[fontstack].ranges[range] = data.stacks[0];
+                for (let i = 0; i < missingRanges[range].length; i++) {
+                    const glyphID = missingRanges[range][i];
+                    const glyph = stack.glyphs[glyphID];
+                    const rect  = atlas.addGlyph(uid, fontstack, glyph, buffer);
+                    if (glyph && rect) glyphs[glyphID] = new SimpleGlyph(glyph, rect, buffer);
```

Comment:
Checking for `rect` is causing some rendering tests to fail due to a missing space between words -- looks like maybe we rely on `rect` being optional in the case of a space character?  (I don't know enough about how we're managing glyph data to know what makes the most sense here.)